### PR TITLE
[SPARK-42842][SQL] Merge the error class _LEGACY_ERROR_TEMP_2006 into REGEX_GROUP_INDEX

### PR DESCRIPTION
**Pull request by liang3zy22**
### What changes were proposed in this pull request?

Merge the error class _LEGACY_ERROR_TEMP_2006 into REGEX_GROUP_INDEX.

### Why are the changes needed?

Fix jira issue [SPARK-42842](https://issues.apache.org/jira/browse/SPARK-42842). The original name just a number, update it to an informal name.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

Current tests covered it.

### CODE REVIEW

File: core/src/main/resources/error/error-classes.json [hunk 1]
```diff
@@ -1009,6 +1009,11 @@
           "Initial bytes from input <saltedMagic> do not match 'Salted__' (0x53616C7465645F5F)."
         ]
       },
+      "NEGATIVE_REGEX_GROUP_INDEX" : {
```
Comment (MaxGekk) [inline core/src/main/resources/error/error-classes.json:1012]: Can't you re-use the existing error class `REGEX_GROUP_INDEX`?

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/regexpExpressions.scala [hunk 2]
```diff
@@ -748,7 +748,7 @@ object RegExpReplace {
 object RegExpExtractBase {
   def checkGroupIndex(prettyName: String, groupCount: Int, groupIndex: Int): Unit = {
     if (groupIndex < 0) {
-      throw QueryExecutionErrors.regexGroupIndexLessThanZeroError
+      throw QueryExecutionErrors.regexGroupIndexLessThanZeroError(prettyName)
     } else if (groupCount < groupIndex) {
       throw QueryExecutionErrors.regexGroupIndexExceedGroupCountError(
         prettyName, groupCount, groupIndex)
```
Comment (MaxGekk) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/regexpExpressions.scala:754]: How about:
```scala
    if (groupIndex < 0 || groupCount < groupIndex) {
      throw QueryExecutionErrors.invalidRegexGroupIndexError(
        prettyName, groupCount, groupIndex)
    }
```

**Review by MaxGekk — APPROVED** [review]
@liang3zy22 Could you update PR's title and the description according to the recent changes, please.

**Comment by MaxGekk** [comment]
+1, LGTM. Merging to master.
Thank you, @liang3zy22.

**Comment by liang3zy22** [comment]
Don't know why CI "continuous-integration/appveyor/pr" is triggered. Checked other PR, this CI test all not triggered.

**Pull request closed without merging**